### PR TITLE
feat: put headers to the http GET rawOutput

### PR DESCRIPTION
**Pull request by alexey-yarmosh**
Fixes https://github.com/jsdelivr/globalping/issues/347

### CODE REVIEW

File: test/unit/command/http-command.test.ts [hunk 1]
```diff
@@ -282,7 +282,7 @@ describe('http command', () => {
 					},
 					rawHeaders: 'test: abc',
 					rawBody: '200 Ok',
-					rawOutput: '200 Ok',
+					rawOutput: 'HTTP/null 200\ntest: abc\n\n200 Ok',
```
Comment (MartinKolarik) [inline test/unit/command/http-command.test.ts:285]: The test should be improved to be more realistic here. The field should never be `null`

**Review by MartinKolarik — CHANGES_REQUESTED** [review]
Without #136, this sometimes resulted in duplicated output. I suspect it was related to the `respond()` being invoked twice when there are errors. After merging in that PR, I can no longer reproduce that issue but it seems that function still could be called twice in some error scenarios and we should guard against that.

```
HTTP/1.1 200
Server: nginx
Date: Mon, 08 May 2023 17:57:15 GMT
Content-Type: text/html; charset=utf-8
Transfer-Encoding: chunked
Connection: close
Vary: Accept-Encoding
CF-Ray: 7c3416a68a971ca5-AMS
Cache-Control: public, max-age=300, must-revalidate, stale-while-revalidate=600, stale-if-error=86400
ETag: W/"af0c-rinv6EfoRfuH5py1o6W0wHntSB8"
Link: <https://www.jsdelivr.com/>; rel="canonical"
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
Vary: Accept-Encoding
CF-Cache-Status: DYNAMIC
x-render-origin-server: Render
x-response-time: 21ms
Set-Cookie: __cf_bm=q3gL519oGE0P0cyGP8knyRyx9ZnmBFlmfL9d3xi8d20-1683406152-0-ATvh0Q7QEwEdooam/XbNv2+nClUgZko9J9o2JoILkx1soG4+CwymxnxUaVXxmCGm3KRBOxj8eK5+u+zTpxfnoQs=; path=/; expires=Sat, 06-May-23 21:19:12 GMT; domain=.onrender.com; HttpOnly; Secure; SameSite=None
Set-Cookie: _cfuvid=hthV.faQP0_HeTAKGb2E0QHL8Gwyqktol.qE2jrUgao-1683406152786-0-604800000; path=/; domain=.onrender.com; HttpOnly; Secure; SameSite=None
Cache: HIT
X-Cached-Since: 2023-05-06T20:49:12+00:00
X-ID: am3-up-gc88
X-NGINX: nginx-be
Content-Encoding: br
Cache: HIT
X-Cached-Since: 2023-05-08T17:53:10+00:00
X-ID: cec-hw-edge-gc12
X-NGINX: nginx-be

<!DOCTYPE html> <html lang="en"><head><meta charset="utf-8"> <meta content="width=device-width, initial-scale=1" name="viewport"> <meta name="description" content="Optimized for JS and ESM delivery from npm and GitHub. Works with all web formats. Serving more than 150 billion requests per month."> <meta content="jsdelivr, cdn, npm, github, js, javascript, free, open source, oss, deno, wordpress" name="keywords"> <meta content="en_US" property="og:locale"> <meta content="website" property="og:type"> <meta property="og:title" content="jsDelivr - A free, fast, and reliable CDN for JS and Open Source"> <meta property="og:description" content="Optimized for JS and ESM delivery from npm and GitHub. Works with all web formats. Serving more than 150 billion requests per month."> <meta property="og:url" content="https://www.jsdelivr.com"> <meta content="jsDelivr" property="og:site_name"> <meta property="og:image" content="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/img/og-jsdelivr.png"> <meta property="og:image:secure_url" content="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/img/og-jsdelivr.png"> <meta content="1280" property="og:image:width"> <meta content="640" property="og:image:height"> <meta name="twitter:image" content="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/img/og-jsdelivr.png"> <meta content="summary_large_image" name="twitter:card"> <meta name="twitter:description" content="Optimized for JS and ESM delivery from npm and GitHub. Works with all web formats. Serving more than 150 billion requests per month."> <meta name="twitter:title" content="jsDelivr - A free, fast, and reliable CDN for JS and Open Source"> <meta content="@jsdelivr" name="twitter:site"> <meta content="@jsdelivr" name="twitter:creator"> <title>jsDelivr - A free, fast, and reliable CDN for JS and open source</title> <script type="application/ld+json">		[
			{
				"@context": "http://schema.org",
				"@type": "Organization",
				"url": "https://www.jsdelivr.com/",
				"sameAs": [
					"https://www.linkedin.com/company/jsdelivr/",
					"https://twitter.com/jsdelivr",
					"https://en.wikipedia.org/wiki/JSDelivr"
				],
				"@id": "#organization",
				"name": "jsDelivr",
				"alternateName": "JSDelivr",
				"foundingDate": "2012-04-10",
				"logo": "https://www.jsdelivr.com/icon_256x256.png",
				"founders": [
					"Dmitriy Akulov"
				],
				"numberOfEmployees": "5"
			},
			{
				"@context": "https://schema.org",
				"@type": "WebSite",
				"url": "https://www.jsdelivr.com/",
				"potentialAction": {
					"@type": "SearchAction",
					"target": {
						"@type": "EntryPoint",
						"urlTemplate": "https://www.jsdelivr.com/?query={search_term_string}"
					},
					"query-input": "required name=search_term_string"
				}
			}
		]
	</script> <!--  Maybe will be useful in Firefox at some point.  --> <link href="https://cdn.jsdelivr.net/" rel="dns-prefetch"> <link href="https://fonts.gstatic.com/" rel="dns-prefetch"> <link href="https://data.jsdelivr.com/" rel="dns-prefetch"> <link href="https://datum.jsdelivr.com/" rel="dns-prefetch"> <link crossorigin="anonymous" href="https://cdn.jsdelivr.net/" rel="preconnect"> <link crossorigin="anonymous" href="https://fonts.gstatic.com/" rel="preconnect"> <link crossorigin="anonymous" href="https://data.jsdelivr.com/" rel="preconnect"> <link crossorigin="anonymous" href="https://datum.jsdelivr.com/" rel="preconnect"> <link crossorigin="anonymous" href="https://fonts.googleapis.com" rel="preconnect"> <link onload="this.media=&#39;all&#39;" media="print" href="https://fonts.googleapis.com/css2?family=Lexend:wght@400;500;600&amp;display=swap" rel="stylesheet"> <link onload="this.media=&#39;all&#39;" media="print" href="https://fonts.googleapis.com/css2?family=Source+Code+Pro:wght@500&amp;display=swap" rel="stylesheet"> <link rel="stylesheet" href="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/css/app.css?v=bc28555c6b206c75a7fdded4e6a2e69b19ea2170"> <link onload="this.media=&#39;all&#39;" media="print" href="https://cdn.jsdelivr.net/npm/font-awesome@4.7.0/css/font-awesome.min.css" rel="stylesheet"> <link onload="this.media=&#39;all&#39;" media="print" href="https://fonts.googleapis.com/css2?family=Inter&amp;display=swap" rel="stylesheet"> <link type="application/xml" href="/sitemap/index.xml" rel="sitemap"> <link title="jsDelivr" type="application/opensearchdescription+xml" href="/opensearch.xml" rel="search"> </head> <body><div id="page">  <header class="c-header  "><nav class="navbar navbar-default "><div class="container-fluid container-mw"><div class="navbar-header"><a class="main-logo" href="/"><!-- 
						Looking for the logo?
						Get all versions here: https://github.com/jsdelivr/jsdelivr-media
					 -->  <img title="jsDelivr" height="34" width="140" src="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/img/jsdelivr-horizontal-regular.svg"></a>   <button class="navbar-toggle collapsed" aria-expanded="false" data-target="#main-navbar" data-toggle="collapse" type="button"><span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span></button></div> <div class="collapse navbar-collapse" id="main-navbar"><ul class="nav navbar-nav navbar-right default-navbar-right"><!-- 					<li><a href="/globalping">Globalping <span class="nav-badge-new">NEW</span></a></li> --> <li><a href="/about">About Us</a></li> <li><a href="/network">Network</a></li> <li><a href="/statistics">Stats</a></li> <li><a href="/sponsors">Sponsors</a></li> <li class="dropdown"><a class="dropdown-toggle" aria-expanded="false" aria-haspopup="true" role="button" data-toggle="dropdown">Tools <i class="fa fa-angle-down" aria-hidden="true"></i></a> <ul class="dropdown-menu"><li><a href="/tools/purge">Purge cache</a></li> <li class="divider" role="separator"></li> <li class="dropdown-header"><div>Convert from</div></li> <li><a href="/github">GitHub</a></li> <li><a href="/google">Google Hosted Libraries</a></li> <li><a href="/unpkg">unpkg</a></li> <li><a href="/skypack">Skypack</a></li> <li><a href="/esmsh">esm.sh</a></li></ul></li> <li class="dropdown"><a class="dropdown-toggle" aria-expanded="false" aria-haspopup="true" role="button" data-toggle="dropdown">Resources <i class="fa fa-angle-down" aria-hidden="true"></i></a> <ul class="dropdown-menu"><li><a class="router-ignore" href="/blog/" target="_blank">Blog <i class="fa fa-external-link" aria-hidden="true"></i></a></li> <li><a href="https://jsdelivr.us11.list-manage.com/subscribe?u=0c85fe2c9a60310df9a3d6751&amp;id=283887f736" rel="noopener noreferrer" target="_blank">Newsletter <i class="fa fa-external-link" aria-hidden="true"></i></a></li> <li><a href="/documentation">Documentation</a></li> <li><a href="/esm">esm.run</a></li> <li><a class="router-ignore" href="/docs/data.jsdelivr.com">API</a></li></ul></li> <li><a title="GitHub" href="https://github.com/jsdelivr/jsdelivr" rel="noopener noreferrer" target="_blank"><i class="fa fa-github" aria-hidden="true"></i> <span class="icon-label-mobile">Github</span></a></li> <li><a title="Twitter" href="https://twitter.com/jsDelivr" rel="noopener noreferrer" target="_blank"><i class="fa fa-twitter" aria-hidden="true"></i> <span class="icon-label-mobile">Twitter</span></a></li> <li><a class="router-ignore" title="Discord" href="/discord" rel="noopener noreferrer" target="_blank"><svg class="fa" style="margin-top: 1px;" title="Discord" height="14" width="14" viewBox="0 0 640 512" xmlns="http://www.w3.org/2000/svg"><path d="M524.531 69.836a1.5 1.5 0 0 0-.764-.7A485.065 485.065 0 0 0 404.081 32.03a1.816 1.816 0 0 0-1.923.91 337.461 337.461 0 0 0-14.9 30.6 447.848 447.848 0 0 0-134.426 0 309.541 309.541 0 0 0-15.135-30.6 1.89 1.89 0 0 0-1.924-.91 483.689 483.689 0 0 0-119.688 37.107 1.712 1.712 0 0 0-.788.676C39.068 183.651 18.186 294.69 28.43 404.354a2.016 2.016 0 0 0 .765 1.375 487.666 487.666 0 0 0 146.825 74.189 1.9 1.9 0 0 0 2.063-.676A348.2 348.2 0 0 0 208.12 430.4a1.86 1.86 0 0 0-1.019-2.588 321.173 321.173 0 0 1-45.868-21.853 1.885 1.885 0 0 1-.185-3.126 251.047 251.047 0 0 0 9.109-7.137 1.819 1.819 0 0 1 1.9-.256c96.229 43.917 200.41 43.917 295.5 0a1.812 1.812 0 0 1 1.924.233 234.533 234.533 0 0 0 9.132 7.16 1.884 1.884 0 0 1-.162 3.126 301.407 301.407 0 0 1-45.89 21.83 1.875 1.875 0 0 0-1 2.611 391.055 391.055 0 0 0 30.014 48.815 1.864 1.864 0 0 0 2.063.7A486.048 486.048 0 0 0 610.7 405.729a1.882 1.882 0 0 0 .765-1.352c12.264-126.783-20.532-236.912-86.934-334.541ZM222.491 337.58c-28.972 0-52.844-26.587-52.844-59.239s23.409-59.241 52.844-59.241c29.665 0 53.306 26.82 52.843 59.239 0 32.654-23.41 59.241-52.843 59.241Zm195.38 0c-28.971 0-52.843-26.587-52.843-59.239s23.409-59.241 52.843-59.241c29.667 0 53.307 26.82 52.844 59.239 0 32.654-23.177 59.241-52.844 59.241Z" fill="#17233b"></path></svg> <span class="icon-label-mobile">Discord</span></a></li></ul></div></div></nav></header> <div class="home"><div class="c-header-gradient"><img class="header-gradient_primary" height="684" width="1440" fetchpriority="high" src="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/img/about/about-gradient.jpg"> <div class="header-gradient_secondary"></div> <div class="header-gradient_map"></div></div> <div class="p-index page"><div class="c-landing-top"><div class="content"><section class="heading"><span>Fast. Reliable. Automated.</span> <h1>A free CDN for open source projects</h1> <p>Optimized for JS and ESM delivery from npm and GitHub. Works with all web formats.</p> <span>Operating since 2012.</span></section> <section class="stats"><div><h3>150 billion requests</h3> <span>past month</span></div> <div><h3>over 5 000 TB</h3> <span>past month</span></div></section></div> <div class="container-fluid container-mw"><div class="row"><div class="col-xs-12 col-lg-10 col-lg-push-1"><section class="docs"><div class="docs_tabs"><div class="docs_tabs_wrapper"><div class="docs_tabs_tab active-docs-tab"><a class="active"><img height="14" width="50" src="/assets/bc28555c6b206c75a7fdded4e6aHTTP/1.1 200
Server: nginx
Date: Mon, 08 May 2023 17:57:15 GMT
Content-Type: text/html; charset=utf-8
Transfer-Encoding: chunked
Connection: close
Vary: Accept-Encoding
CF-Ray: 7c3416a68a971ca5-AMS
Cache-Control: public, max-age=300, must-revalidate, stale-while-revalidate=600, stale-if-error=86400
ETag: W/"af0c-rinv6EfoRfuH5py1o6W0wHntSB8"
Link: <https://www.jsdelivr.com/>; rel="canonical"
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
Vary: Accept-Encoding
CF-Cache-Status: DYNAMIC
x-render-origin-server: Render
x-response-time: 21ms
Set-Cookie: __cf_bm=q3gL519oGE0P0cyGP8knyRyx9ZnmBFlmfL9d3xi8d20-1683406152-0-ATvh0Q7QEwEdooam/XbNv2+nClUgZko9J9o2JoILkx1soG4+CwymxnxUaVXxmCGm3KRBOxj8eK5+u+zTpxfnoQs=; path=/; expires=Sat, 06-May-23 21:19:12 GMT; domain=.onrender.com; HttpOnly; Secure; SameSite=None
Set-Cookie: _cfuvid=hthV.faQP0_HeTAKGb2E0QHL8Gwyqktol.qE2jrUgao-1683406152786-0-604800000; path=/; domain=.onrender.com; HttpOnly; Secure; SameSite=None
Cache: HIT
X-Cached-Since: 2023-05-06T20:49:12+00:00
X-ID: am3-up-gc88
X-NGINX: nginx-be
Content-Encoding: br
Cache: HIT
X-Cached-Since: 2023-05-08T17:53:10+00:00
X-ID: cec-hw-edge-gc12
X-NGINX: nginx-be

<!DOCTYPE html> <html lang="en"><head><meta charset="utf-8"> <meta content="width=device-width, initial-scale=1" name="viewport"> <meta name="description" content="Optimized for JS and ESM delivery from npm and GitHub. Works with all web formats. Serving more than 150 billion requests per month."> <meta content="jsdelivr, cdn, npm, github, js, javascript, free, open source, oss, deno, wordpress" name="keywords"> <meta content="en_US" property="og:locale"> <meta content="website" property="og:type"> <meta property="og:title" content="jsDelivr - A free, fast, and reliable CDN for JS and Open Source"> <meta property="og:description" content="Optimized for JS and ESM delivery from npm and GitHub. Works with all web formats. Serving more than 150 billion requests per month."> <meta property="og:url" content="https://www.jsdelivr.com"> <meta content="jsDelivr" property="og:site_name"> <meta property="og:image" content="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/img/og-jsdelivr.png"> <meta property="og:image:secure_url" content="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/img/og-jsdelivr.png"> <meta content="1280" property="og:image:width"> <meta content="640" property="og:image:height"> <meta name="twitter:image" content="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/img/og-jsdelivr.png"> <meta content="summary_large_image" name="twitter:card"> <meta name="twitter:description" content="Optimized for JS and ESM delivery from npm and GitHub. Works with all web formats. Serving more than 150 billion requests per month."> <meta name="twitter:title" content="jsDelivr - A free, fast, and reliable CDN for JS and Open Source"> <meta content="@jsdelivr" name="twitter:site"> <meta content="@jsdelivr" name="twitter:creator"> <title>jsDelivr - A free, fast, and reliable CDN for JS and open source</title> <script type="application/ld+json">		[
			{
				"@context": "http://schema.org",
				"@type": "Organization",
				"url": "https://www.jsdelivr.com/",
				"sameAs": [
					"https://www.linkedin.com/company/jsdelivr/",
					"https://twitter.com/jsdelivr",
					"https://en.wikipedia.org/wiki/JSDelivr"
				],
				"@id": "#organization",
				"name": "jsDelivr",
				"alternateName": "JSDelivr",
				"foundingDate": "2012-04-10",
				"logo": "https://www.jsdelivr.com/icon_256x256.png",
				"founders": [
					"Dmitriy Akulov"
				],
				"numberOfEmployees": "5"
			},
			{
				"@context": "https://schema.org",
				"@type": "WebSite",
				"url": "https://www.jsdelivr.com/",
				"potentialAction": {
					"@type": "SearchAction",
					"target": {
						"@type": "EntryPoint",
						"urlTemplate": "https://www.jsdelivr.com/?query={search_term_string}"
					},
					"query-input": "required name=search_term_string"
				}
			}
		]
	</script> <!--  Maybe will be useful in Firefox at some point.  --> <link href="https://cdn.jsdelivr.net/" rel="dns-prefetch"> <link href="https://fonts.gstatic.com/" rel="dns-prefetch"> <link href="https://data.jsdelivr.com/" rel="dns-prefetch"> <link href="https://datum.jsdelivr.com/" rel="dns-prefetch"> <link crossorigin="anonymous" href="https://cdn.jsdelivr.net/" rel="preconnect"> <link crossorigin="anonymous" href="https://fonts.gstatic.com/" rel="preconnect"> <link crossorigin="anonymous" href="https://data.jsdelivr.com/" rel="preconnect"> <link crossorigin="anonymous" href="https://datum.jsdelivr.com/" rel="preconnect"> <link crossorigin="anonymous" href="https://fonts.googleapis.com" rel="preconnect"> <link onload="this.media=&#39;all&#39;" media="print" href="https://fonts.googleapis.com/css2?family=Lexend:wght@400;500;600&amp;display=swap" rel="stylesheet"> <link onload="this.media=&#39;all&#39;" media="print" href="https://fonts.googleapis.com/css2?family=Source+Code+Pro:wght@500&amp;display=swap" rel="stylesheet"> <link rel="stylesheet" href="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/css/app.css?v=bc28555c6b206c75a7fdded4e6a2e69b19ea2170"> <link onload="this.media=&#39;all&#39;" media="print" href="https://cdn.jsdelivr.net/npm/font-awesome@4.7.0/css/font-awesome.min.css" rel="stylesheet"> <link onload="this.media=&#39;all&#39;" media="print" href="https://fonts.googleapis.com/css2?family=Inter&amp;display=swap" rel="stylesheet"> <link type="application/xml" href="/sitemap/index.xml" rel="sitemap"> <link title="jsDelivr" type="application/opensearchdescription+xml" href="/opensearch.xml" rel="search"> </head> <body><div id="page">  <header class="c-header  "><nav class="navbar navbar-default "><div class="container-fluid container-mw"><div class="navbar-header"><a class="main-logo" href="/"><!-- 
						Looking for the logo?
						Get all versions here: https://github.com/jsdelivr/jsdelivr-media
					 -->  <img title="jsDelivr" height="34" width="140" src="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/img/jsdelivr-horizontal-regular.svg"></a>   <button class="navbar-toggle collapsed" aria-expanded="false" data-target="#main-navbar" data-toggle="collapse" type="button"><span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span></button></div> <div class="collapse navbar-collapse" id="main-navbar"><ul class="nav navbar-nav navbar-right default-navbar-right"><!-- 					<li><a href="/globalping">Globalping <span class="nav-badge-new">NEW</span></a></li> --> <li><a href="/about">About Us</a></li> <li><a href="/network">Network</a></li> <li><a href="/statistics">Stats</a></li> <li><a href="/sponsors">Sponsors</a></li> <li class="dropdown"><a class="dropdown-toggle" aria-expanded="false" aria-haspopup="true" role="button" data-toggle="dropdown">Tools <i class="fa fa-angle-down" aria-hidden="true"></i></a> <ul class="dropdown-menu"><li><a href="/tools/purge">Purge cache</a></li> <li class="divider" role="separator"></li> <li class="dropdown-header"><div>Convert from</div></li> <li><a href="/github">GitHub</a></li> <li><a href="/google">Google Hosted Libraries</a></li> <li><a href="/unpkg">unpkg</a></li> <li><a href="/skypack">Skypack</a></li> <li><a href="/esmsh">esm.sh</a></li></ul></li> <li class="dropdown"><a class="dropdown-toggle" aria-expanded="false" aria-haspopup="true" role="button" data-toggle="dropdown">Resources <i class="fa fa-angle-down" aria-hidden="true"></i></a> <ul class="dropdown-menu"><li><a class="router-ignore" href="/blog/" target="_blank">Blog <i class="fa fa-external-link" aria-hidden="true"></i></a></li> <li><a href="https://jsdelivr.us11.list-manage.com/subscribe?u=0c85fe2c9a60310df9a3d6751&amp;id=283887f736" rel="noopener noreferrer" target="_blank">Newsletter <i class="fa fa-external-link" aria-hidden="true"></i></a></li> <li><a href="/documentation">Documentation</a></li> <li><a href="/esm">esm.run</a></li> <li><a class="router-ignore" href="/docs/data.jsdelivr.com">API</a></li></ul></li> <li><a title="GitHub" href="https://github.com/jsdelivr/jsdelivr" rel="noopener noreferrer" target="_blank"><i class="fa fa-github" aria-hidden="true"></i> <span class="icon-label-mobile">Github</span></a></li> <li><a title="Twitter" href="https://twitter.com/jsDelivr" rel="noopener noreferrer" target="_blank"><i class="fa fa-twitter" aria-hidden="true"></i> <span class="icon-label-mobile">Twitter</span></a></li> <li><a class="router-ignore" title="Discord" href="/discord" rel="noopener noreferrer" target="_blank"><svg class="fa" style="margin-top: 1px;" title="Discord" height="14" width="14" viewBox="0 0 640 512" xmlns="http://www.w3.org/2000/svg"><path d="M524.531 69.836a1.5 1.5 0 0 0-.764-.7A485.065 485.065 0 0 0 404.081 32.03a1.816 1.816 0 0 0-1.923.91 337.461 337.461 0 0 0-14.9 30.6 447.848 447.848 0 0 0-134.426 0 309.541 309.541 0 0 0-15.135-30.6 1.89 1.89 0 0 0-1.924-.91 483.689 483.689 0 0 0-119.688 37.107 1.712 1.712 0 0 0-.788.676C39.068 183.651 18.186 294.69 28.43 404.354a2.016 2.016 0 0 0 .765 1.375 487.666 487.666 0 0 0 146.825 74.189 1.9 1.9 0 0 0 2.063-.676A348.2 348.2 0 0 0 208.12 430.4a1.86 1.86 0 0 0-1.019-2.588 321.173 321.173 0 0 1-45.868-21.853 1.885 1.885 0 0 1-.185-3.126 251.047 251.047 0 0 0 9.109-7.137 1.819 1.819 0 0 1 1.9-.256c96.229 43.917 200.41 43.917 295.5 0a1.812 1.812 0 0 1 1.924.233 234.533 234.533 0 0 0 9.132 7.16 1.884 1.884 0 0 1-.162 3.126 301.407 301.407 0 0 1-45.89 21.83 1.875 1.875 0 0 0-1 2.611 391.055 391.055 0 0 0 30.014 48.815 1.864 1.864 0 0 0 2.063.7A486.048 486.048 0 0 0 610.7 405.729a1.882 1.882 0 0 0 .765-1.352c12.264-126.783-20.532-236.912-86.934-334.541ZM222.491 337.58c-28.972 0-52.844-26.587-52.844-59.239s23.409-59.241 52.844-59.241c29.665 0 53.306 26.82 52.843 59.239 0 32.654-23.41 59.241-52.843 59.241Zm195.38 0c-28.971 0-52.843-26.587-52.843-59.239s23.409-59.241 52.843-59.241c29.667 0 53.307 26.82 52.844 59.239 0 32.654-23.177 59.241-52.844 59.241Z" fill="#17233b"></path></svg> <span class="icon-label-mobile">Discord</span></a></li></ul></div></div></nav></header> <div class="home"><div class="c-header-gradient"><img class="header-gradient_primary" height="684" width="1440" fetchpriority="high" src="/assets/bc28555c6b206c75a7fdded4e6a2e69b19ea2170/img/about/about-gradient.jpg"> <div class="header-gradient_secondary"></div> <div class="header-gradient_map"></div></div> <div class="p-index page"><div class="c-landing-top"><div class="content"><section class="heading"><span>Fast. Reliable. Automated.</span> <h1>A free CDN for open source projects</h1> <p>Optimized for JS and ESM delivery from npm and GitHub. Works with all web formats.</p> <span>Operating since 2012.</span></section> <section class="stats"><div><h3>150 billion requests</h3> <span>past month</span></div> <div><h3>over 5 000 TB</h3> <span>past month</span></div></section></div> <div class="container-fluid container-mw"><div class="row"><div class="col-xs-12 col-lg-10 col-lg-push-1"><section class="docs"><div class="docs_tabs"><div class="docs_tabs_wrapper"><div class="docs_tabs_tab active-docs-tab"><a class="active"><img height="14" width="50" src="/assets/bc28555c6b206c75a7fdded4e6a
```

**Comment by alexey-yarmosh** [comment]
Probably the issue with the double output was related to additional `stream.on('close', () => { respond(resolve); });` in the early version of PR. Respond was called on error and then on 'end' event. Now the issue is not reproducible.

File: src/command/http-command.ts [hunk 2]
```diff
@@ -93,18 +93,18 @@ const getInitialResult = () => ({
 	timings: {},
 });
 
-const allowedHttpProtocols = [ 'http', 'https', 'http2' ];
-const allowedHttpMethods = [ 'get', 'head' ];
+const allowedHttpProtocols = [ 'HTTP', 'HTTPS', 'HTTP2' ];
+const allowedHttpMethods = [ 'GET', 'HEAD' ];
 
 export const httpOptionsSchema = Joi.object<HttpOptions>({
-	type: Joi.string().valid('http').insensitive().required(),
```
Comment (MartinKolarik) [inline src/command/http-command.ts:100]: The fields are publicly case-insensitive. In theory, the are normalized to uppercase by the API, but it still seems safer to keep them insensitive here as well? Joi will correctly normalize them to uppercase if the uppercased versions are passed to `valid()`

Comment (MartinKolarik) [reply src/command/http-command.ts:100]: Was it actually wrong? Joi normalizes the value, so until now, it was converting everything to lowercase thanks to the `.insensitive()` part - that's why I was able to change what API does without breaking the probe communication.

Comment (alexey-yarmosh) [reply src/command/http-command.ts:100]: You are right, seems it is converting correctly. Ok, lets keep the `insensitive` part.

**Review by MartinKolarik — APPROVED** [review]

**Pull request merged**